packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -2885,11 +3091,131 @@ function remountFiber(
   }
 }
 
+function reifyWorkInProgress(current: Fiber, workInProgress: Fiber | null) {
```

Comment:
This is the heavy lift, it is basically the child cloning algo but done in reverse walking back up the return list to get to a state we would have been in if speculative work was not a feature